src/pykeen/evaluation/inductive_evaluator.py

```diff
@@ -0,0 +1,489 @@
+# -*- coding: utf-8 -*-
+
+"""Inductive evaluator."""
+
+import gc
+import logging
+import timeit
+from abc import ABC, abstractmethod
+from contextlib import contextmanager
+from dataclasses import dataclass
+from math import ceil
+from textwrap import dedent
+from typing import Any, Collection, Iterable, List, Mapping, Optional, Tuple, Union, cast
+
+import numpy as np
+import torch
+from dataclasses_json import DataClassJsonMixin
+from tqdm.autonotebook import tqdm
+
+from .evaluator import (
+    Evaluator,
+    MetricResults,
+    create_dense_positive_mask_,
+    create_sparse_positive_filter_,
+    filter_scores_,
+)
+from .rank_based_evaluator import RankBasedEvaluator
+from ..models import Model
+from ..triples.triples_factory import TriplesFactory, restrict_triples
+from ..triples.utils import get_entities, get_relations
+from ..typing import MappedTriples
+from ..utils import (
+    format_relative_comparison,
+    is_cuda_oom_error,
+    is_cudnn_error,
+    is_nonzero_larger_than_maxint_error,
+    normalize_string,
+    split_list_in_batches_iter,
+)
+
+__all__ = [
+    "InductiveEvaluator",
+]
+
+logger = logging.getLogger(__name__)
+
+
+@contextmanager
+def optional_context_manager(condition, context_manager):
+    if condition:
+        with context_manager:
+            yield context_manager
+    else:
+        yield
+
+
+class InductiveEvaluator(RankBasedEvaluator):
+    """
+    Inductive version of the evaluator. Main differences:
+    - Takes the triple factory argument - on which the evaluation will be executed
+    - Takes the mode argument which will be sent to the scoring function
+    """
+
+    def __init__(self, eval_factory: TriplesFactory, mode: str, **kwargs):
+        super().__init__(**kwargs)
+
+        self.mode = mode
+        self.eval_factory = eval_factory
```

Comment:
where is this instance variable used? Can it be removed?

Reply:
Yes this can be removed - EarlyStopper takes the necessary arguments for filtering upon construction, and at test time those factories can be sent directly to the `evaluate()` function